nbt/src/main/java/net/kyori/adventure/nbt/BinaryTagIO.java

```diff
@@ -182,11 +200,283 @@ public static void writeCompressedOutputStream(final @NonNull CompoundBinaryTag
+
+    /**
+     * Writes a binary tag, with a name, to {@code path}.
+     *
+     * @param path the path
+     * @throws IOException if an exception was encountered while reading the tag
+     * @since 4.4.0
+     */
+    default void writeNamed(final Map.@NonNull Entry<String, CompoundBinaryTag> tag, final @NonNull Path path) throws IOException {
+      this.writeNamed(tag, path, null);
+    }
+
+    /**
+     * Writes a binary tag, with a name, to {@code path} with a {@code compression} type.
+     *
+     * <p>No compression will be used if {@code compression} is {@code null}.</p>
+     *
+     * @param path the path
+     * @param compression the compression type
+     * @throws IOException if an exception was encountered while reading the tag
+     * @since 4.4.0
+     */
+    void writeNamed(final Map.@NonNull Entry<String, CompoundBinaryTag> tag, final @NonNull Path path, final @Nullable Compression compression) throws IOException;
+
+    /**
+     * Writes a binary tag, with a name, to {@code output}.
+     *
+     * @param output the output stream
+     * @throws IOException if an exception was encountered while reading the tag
+     * @since 4.4.0
+     */
+    default void writeNamed(final Map.@NonNull Entry<String, CompoundBinaryTag> tag, final @NonNull OutputStream output) throws IOException {
+      this.writeNamed(tag, output, null);
+    }
+
+    /**
+     * Writes a binary tag, with a name, to {@code output} with a {@code compression} type.
+     *
+     * <p>No compression will be used if {@code compression} is {@code null}.</p>
+     *
+     * @param output the output stream
+     * @param compression the compression type
+     * @throws IOException if an exception was encountered while reading the tag
+     * @since 4.4.0
+     */
+    void writeNamed(final Map.@NonNull Entry<String, CompoundBinaryTag> tag, final @NonNull OutputStream output, final @Nullable Compression compression) throws IOException;
+
+    /**
+     * Writes a binary tag, with a name, to {@code output}.
+     *
+     * @param output the output
+     * @throws IOException if an exception was encountered while reading the tag
+     * @since 4.4.0
+     */
+    void writeNamed(final Map.@NonNull Entry<String, CompoundBinaryTag> tag, final @NonNull DataOutput output) throws IOException;
+  }
+
+  /**
+   * Compression types.
+   *
+   * @since 4.4.0
+   */
+  public static abstract class Compression {
+    /**
+     * GZIP compression.
+     *
+     * @since 4.4.0
+     */
+    public static final Compression GZIP = new Compression() {
+      @Override
+      @NonNull InputStream decompress(final @NonNull InputStream is) throws IOException {
+        return new GZIPInputStream(is);
+      }
+
+      @Override
+      @NonNull OutputStream compress(final @NonNull OutputStream os) throws IOException {
+        return new GZIPOutputStream(os);
+      }
+    };
+
+    abstract @NonNull InputStream decompress(final @NonNull InputStream is) throws IOException;
```

Comment:
these methods should have JD, looks like.

i assume your intent here is others could implement Compression to write NBT with some other format, rather than gzip?

Reply:
These two methods are package-private

Reply:
ah -- i've spent too much time looking at interfaces i gues :)